### PR TITLE
[search] Get rid of gradient descent, reuse linear svm.

### PR DESCRIPTION
**Pull request by ygorshenin**

### CODE REVIEW

**Comment by gardster** [comment]
LGTM

**Comment by ygorshenin** [comment]
@mpimenov PTAL

File: search/search_quality/scoring_model.py [hunk 1]
```diff
@@ -64,121 +79,62 @@ def compute_ndcg_for_w(data, w):
...
+    clf = svm.LinearSVC(random_state=args.seed)
+    cv = cross_validation.KFold(len(y), n_folds=5, shuffle=True, random_state=args.seed)
 
-    def gradient(self, w):
-        result = 2 * self.lam * w
-        for coeff in self.coeffs:
-            if 1 - np.dot(coeff, w) > 0:
-                result = result - coeff
-        return result
+    # Let's find regularizer by a grid search.
```
Comment (mpimenov) [inline search/search_quality/scoring_model.py]: Let's not write useless comments.

File: search/search_quality/scoring_model.py [hunk 2]
```diff
@@ -64,121 +79,62 @@ def compute_ndcg_for_w(data, w):
...
+    xs, ys = [], []
+    for id in grouped:
+        indices = grouped[id]
+        features = data.ix[indices][FEATURES]
+        relevances = np.array(data.ix[indices]['Relevance'])
 
-    def __init__(self, data, lam=1e-3):
-        self.coeffs, self.lam = [], lam
+        n, total = len(indices), 0
+        for k, (i, j) in enumerate(itertools.combinations(range(n), 2)):
```
Comment (mpimenov) [inline search/search_quality/scoring_model.py]: s/k/_/?

File: search/search_quality/scoring_model.py (hunk 1 above)
Comment (ygorshenin) [inline search/search_quality/scoring_model.py]: It's not obvious that C is a regularizer constant. But if you insist, I'll remove the comment.

Comment (deathbaba) [inline search/search_quality/scoring_model.py]: Hey, it's useful.

Comment (mpimenov) [inline search/search_quality/scoring_model.py]: Why not write "C stands for the regularizer constant" then?

Comment (ygorshenin) [inline search/search_quality/scoring_model.py]: Why not to propose useful comments at once?

**Comment by mpimenov** [comment]
LGTM

File: search/search_quality/scoring_model.py (hunk 1 above)
Comment (ygorshenin) [inline search/search_quality/scoring_model.py]: Done

File: search/search_quality/scoring_model.py (hunk 2 above)
Comment (ygorshenin) [inline search/search_quality/scoring_model.py]: Done

**Comment by ygorshenin** [comment]
Merge, please.

**Comment by ygorshenin** [comment]
@deathbaba ? :)

**Comment by deathbaba** [comment]
Я не питонщик, поэтому LGTM от меня не ждите )

**Comment by ygorshenin** [comment]
Ты его уже написал )

**Comment by ygorshenin** [comment]
Many thanks!

**Pull request merged**